Sources/UAPusher/Builder/UABuilder.swift

```diff
@@ -0,0 +1,46 @@
+import Vapor
+
+/// Builder class for the request payload
+public final class UABuilder {
+    
+    /// Payload segments
+    public var segments: [Segment]
```

Comment:
I think we should make this `private`. We're exposing methods for mutating this, so in order to guide people to use those over using e.g. `append` directly on this variable, it makes sense to have this as `private`. What do you think? One could argue that we could remove `add` and then just make users use `append` always, but I guess our argument is that we simplify the API a little bit as well as getting some control if we at some point want to add functionality (without having to do a breaking change).

Reply:
Let's do that